src/hotspot/os/aix/os_aix.cpp

```diff
@@ -1173,12 +1173,14 @@ void *os::dll_load(const char *filename, char *ebuf, int ebuflen) {
   char* file_path = NEW_C_HEAP_ARRAY(char, buffer_length + extension_length + 1, mtInternal);
   strncpy(file_path,filename, buffer_length + 1);
   char* const pointer_to_dot = strrchr(file_path, '.');
-  assert(pointer_to_dot != nullptr, "Attempting to load a shared object without extension? %s", filename);
+  if (pointer_to_dot == nullptr) {
+    log_info(os)("Attempting to load a shared object without extension %s", filename);
+  }    
   // First try to load the existing file.
   result = dll_load_library(file_path, ebuf, ebuflen);
   // If the load fails,we try to reload by changing the extension to .a for .so files only.
-  if(result == nullptr) {
-    if(strcmp(pointer_to_dot, ".so") == 0) {
+  if (result == nullptr) {
+    if (strcmp(pointer_to_dot, ".so") == 0) {
```

Comment:
We could possibly reach here with `pointer_to_dot` == nullptr. Invoking strcmp causes undefined behavior!

Reply:
Yes. Do we exit when there is no extension in the file ..or we can just append an extension manually and let the program flow go through.